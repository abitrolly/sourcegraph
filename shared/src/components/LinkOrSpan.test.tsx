import React from 'react'
import { LinkOrSpan } from './LinkOrSpan'
import { mount } from 'enzyme'

describe('LinkOrSpan', () => {
    test('render a link when "to" is set', () => {
        expect(mount(<LinkOrSpan to="http://example.com">foo</LinkOrSpan>)).toMatchSnapshot()
    })

    test('render a span when "to" is undefined', () => {
        expect(mount(<LinkOrSpan to={undefined}>foo</LinkOrSpan>)).toMatchSnapshot()
    })
})
