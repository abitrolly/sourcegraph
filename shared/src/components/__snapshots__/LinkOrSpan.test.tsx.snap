// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkOrSpan render a link when "to" is set 1`] = `
<LinkOrSpan
  to="http://example.com"
>
  <AnchorLink
    className=""
    to="http://example.com"
  >
    <a
      className=""
      href="http://example.com"
    >
      foo
    </a>
  </AnchorLink>
</LinkOrSpan>
`;

exports[`LinkOrSpan render a span when "to" is undefined 1`] = `
<LinkOrSpan>
  <span
    className=""
  >
    foo
  </span>
</LinkOrSpan>
`;
